src/SFA.DAS.ToolsNotifications.Api/Startup.cs

```diff
@@ -1,41 +1,115 @@
+using Microsoft.OpenApi.Models;
+using SFA.DAS.ToolsNotifications.Api.Infrastructure;
 using SFA.DAS.ToolsNotifications.Core.Repositories;
 using SFA.DAS.ToolsNotifications.Core.Services;
 using SFA.DAS.ToolsNotifications.Infrastructure.Repositories;
+using System;
+using System.Collections.Generic;
 
 namespace SFA.DAS.ToolsNotifications.Api
 {
     public class Startup
     {
-        public IConfiguration Configuration { get; }
-        private IHostingEnvironment CurrentEnvironment { get; }
+        private readonly IConfiguration _configuration;
+        private readonly IHostingEnvironment _environment;
 
-        public Startup(IConfiguration configuration, IHostingEnvironment env)
+        public Startup(IConfiguration configuration, IHostingEnvironment environment)
         {
-            Configuration = configuration;
-            CurrentEnvironment = env;
+            _environment = environment;
+            _configuration = configuration;
         }
 
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+            services.AddApplicationInsightsTelemetry(_configuration["APPINSIGHTS_INSTRUMENTATIONKEY"]);
 
-            if (CurrentEnvironment.IsDevelopment())
+            if (!ConfigurationIsLocalOrDev())
             {
+                services.AddAuthentication(auth => { auth.DefaultScheme = JwtBearerDefaults.AuthenticationScheme; })
+                    .AddJwtBearer(auth =>
+                    {
+                        auth.Authority =
+                            $"https://login.microsoftonline.com/{_configuration["AzureAdTenantId"]}";
+                        auth.TokenValidationParameters = new Microsoft.IdentityModel.Tokens.TokenValidationParameters
+                        {
+                            ValidAudiences = new List<string>
+                            {
+                            _configuration["AzureADResourceId"],
+                            _configuration["AzureADClientId"]
+                            }
+                        };
+                    });
+
+                services.AddSingleton<IClaimsTransformation, AzureAdScopeClaimTransformation>();
+
+                services.AddAuthorization(options =>
+                {
+                    options.AddPolicy("RequireNotificationRole", policy =>
+                    {
+                        policy.RequireAuthenticatedUser();
+                        policy.RequireRole("Notifications");
+                    });
+                });
+
                 services.AddDistributedMemoryCache();
             }
             else
             {
-                services.AddStackExchangeRedisCache(options => options.Configuration = Configuration["RedisConnectionString"]);
+                services.AddStackExchangeRedisCache(options => options.Configuration = _configuration["RedisConnectionString"]);
```

Comment:
You've flipped the logic in the if statement so now these are the wrong way around